dev/com.ibm.ws.microprofile.faulttolerance_fat_tck/fat/src/com/ibm/ws/microprofile/tck/Utils.java

```diff
@@ -0,0 +1,299 @@
+		// included as the <systemPath> element for the mvn coordinates of the dependency
+		// in tck/pom.xml
+		jarsFromWlp.add("com.ibm.websphere.org.eclipse.microprofile.faulttolerance");
+		jarsFromWlp.add("com.ibm.ws.microprofile.faulttolerance");
+		jarsFromWlp.add("com.ibm.ws.microprofile.faulttolerance.cdi");
+		jarsFromWlp.add("com.ibm.websphere.org.eclipse.microprofile.config");
+		jarsFromWlp.add("com.ibm.ws.microprofile.config");
+		jarsFromWlp.add("com.ibm.ws.microprofile.config.cdi");
+
+		mvnCliRoot = concatStringArray(mvnCliRaw, getJarCliEnvVars(server, jarsFromWlp));
+
+		// The cmd below is a base for running the TCK as a whole for this project.
+		// It is possible to use other Testng control xml files that do useful inclusions
+		// and exclusions and there are some examples (see *.xml) in the tck directory
+		mvnCliTckRoot = concatStringArray(mvnCliRoot, new String[] { "-DsuiteXmlFile=tck-suite.xml" });
+		init = true;
+	}
+
+	/**
+	 * Resolves a set of "-Djarname=path" type strings to add to the CLI. The path
+	 * is resolved to existing jar names that match the jarName but also include
+	 * version numbers etc.
+	 *
+	 * @param server
+	 * @param nonVersionedJars
+	 * @return an array of String that can be added to a ProcessRunner command
+	 */
+	private static String[] getJarCliEnvVars(LibertyServer server, List<String> nonVersionedJars) {
+
+		Map<String, String> actualJarFiles = resolveJarPaths(nonVersionedJars, server);
+		String[] addon = new String[] {};
+
+		Set<String> jarsSet = actualJarFiles.keySet();
+		for (Iterator<String> iterator = jarsSet.iterator(); iterator.hasNext();) {
+			String jarKey = iterator.next();
+			String jarPathName = actualJarFiles.get(jarKey);
+			addon = concatStringArray(addon, new String[] { "-D" + jarKey + "=" + jarPathName });
+		}
+		return addon;
+	}
+
+	/**
+	 * Smashes two String arrays into a returned third. Useful for appending to the
+	 * CLI used for ProcessBuilder. This is not a very efficient way to do this but
+	 * it is easy to read/write and it is not a performance sensitive area.
+	 *
+	 * @param a
+	 * @param b
+	 * @return cat a b
+	 */
+	public static String[] concatStringArray(String[] a, String[] b) {
+		Stream<String> streamA = Arrays.stream(a);
+		Stream<String> streamB = Arrays.stream(b);
+		return Stream.concat(streamA, streamB).toArray(String[]::new);
+	}
+
+	/**
+	 * Run a command using a ProcessBuilder.
+	 *
+	 * @param cmd
+	 * @param workingDirectory
+	 * @param outputFile
+	 * @return The return code of the process. (TCKs return 0 if all tests pass and
+	 *         !=0 otherwise).
+	 * @throws Exception
+	 */
+	public static int runCmd(String[] cmd, File workingDirectory, File outputFile) throws Exception {
+		ProcessBuilder pb = new ProcessBuilder(cmd);
+		pb.directory(workingDirectory);
+		pb.redirectOutput(outputFile);
+		log("Running command " + Arrays.asList(cmd));
+		Process p = pb.start();
+		int exitCode = p.waitFor();
```

Comment:
You should use the version of waitFor which takes a timeout. It could be a very long timeout ... I think the build has a property for the test timeout length.

Reply:
The simplicity framework handles timeout at the top level. I don't want to duplicate that.